library/core/src/fmt/num.rs

```diff
@@ -199,35 +199,45 @@ static DEC_DIGITS_LUT: &[u8; 200] = b"0001020304050607080910111213141516171819\
       8081828384858687888990919293949596979899";
 
 macro_rules! impl_Display {
-    ($($t:ident $(as $positive:ident)? named $name:ident,)* ; as $u:ident via $conv_fn:ident named $gen_name:ident) => {
+    ($($signed:ident, $unsigned:ident,)* ; as $u:ident via $conv_fn:ident named $gen_name:ident) => {
 
         $(
         #[stable(feature = "rust1", since = "1.0.0")]
-        impl fmt::Display for $t {
+        impl fmt::Display for $unsigned {
             fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-                // If it's a signed integer.
-                $(
-                    let is_nonnegative = *self >= 0;
+                #[cfg(not(feature = "optimize_for_size"))]
+                {
+                    self._fmt(true, f)
+                }
+                #[cfg(feature = "optimize_for_size")]
+                {
+                    $gen_name(self.$conv_fn(), true, f)
+                }
+            }
+        }
 
+        #[stable(feature = "rust1", since = "1.0.0")]
+        impl fmt::Display for $signed {
+            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                let is_nonnegative = *self >= 0;
+
+                if !is_nonnegative {
                     #[cfg(not(feature = "optimize_for_size"))]
                     {
-                        if !is_nonnegative {
-                            // convert the negative num to positive by summing 1 to its 2s complement
-                            return (!self as $positive).wrapping_add(1)._fmt(false, f);
-                        }
+                        // convert the negative num to positive by summing 1 to its 2s complement
+                        return (!self as $unsigned).wrapping_add(1)._fmt(false, f);
```

Comment:
Why not simply `wrapping_neg`?

Reply:
Great idea! I didn't realize that it was also converting negative to positive. Gonna need to send a PR to update the doc example.

Reply:
Ok now I now why we didn't use `wrapping_neg`: MIN is MAX + 1 so it cannot work. We need to keep this conversion.

Reply:
That's why we also do the `as` cast.

`i32::MIN.wrapping_neg() == i32::MIN`.

`i32::MIN as u32` returns the corresponding absolute value.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=53b21da1c591da1aae170df06f97e922

Reply:
Ah, I didn't notice you also removed the `as`.

Reply:
Hm, `.wrapping_neg()` has a curious behavior for unsigned integers, didn't expect that.

Reply:
Ah, I hadn't even fully absorbed this is only the signed implementation, which means `unsigned_abs` can be used.